### PR TITLE
Header user language

### PR DESCRIPTION
**Pull request by IkramMaalej**

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java [hunk 1]
```diff
@@ -289,6 +292,10 @@ public void setMetadataLanguage(String metadataLanguage) {
         this.metadataLanguage = metadataLanguage;
     }
 
+    public String getLanguage() { return language; }
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java]: Missing JavaDoc and please adjust formatting of the method.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java [hunk 2]
```diff
@@ -289,6 +292,10 @@ public void setMetadataLanguage(String metadataLanguage) {
...
+
+    public void setLanguage(String language) { this.language = language; }
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java]: Missing JavaDoc and please adjust formatting of the method.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 3]
```diff
@@ -46,16 +50,36 @@
...
-        String key = ConfigCore.getParameter("language.force-default", "de");
+        setSessionLocaleFieldId();
+    }
+
+    /**
+     * Set session locale field id.
+     *
+     *
+     */
+    public void setSessionLocaleFieldId() {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Method must not be public as this method is only used inside this class.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 4]
```diff
@@ -95,6 +119,7 @@ public SpracheForm() {
      * @return a list of maps, each with the fields “id”, “displayName” and
      *         “selected”
      */
+
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Why did you add a empty line here?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 5]
```diff
@@ -112,6 +137,7 @@ public SpracheForm() {
                 translation.put("displayLanguageTranslated",
                         supportedLocale.getDisplayLanguage(currentDisplayLanguage));
                 translation.put("selected", supportedLocale.equals(currentDisplayLanguage));
+                translation.put("flag","javax.faces.resource/images/" + supportedLocale.toString() + ".svg.jsf");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Method parameters are comma separated including a white space.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 6]
```diff
@@ -135,6 +161,15 @@ public void switchLanguage(String langCodeCombined) throws IOException {
         } else {
             locale = new Locale(languageCode[0]);
         }
+        try {
+            User user = serviceManager.getUserService().getById(serviceManager.getUserService().getAuthenticatedUser().getId());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: I think this line is too long. Did you apply the formatting rules?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 7]
```diff
@@ -46,16 +50,36 @@
...
+     *
+     *
+     */
+    public void setSessionLocaleFieldId() {
+        String key = "";
+        if (Objects.isNull(serviceManager.getUserService().getAuthenticatedUser())) {
+            key = ConfigCore.getParameter("language.force-default", "de");
+        } else {
+            try {
+                User user = serviceManager.getUserService().getById(serviceManager.getUserService().getAuthenticatedUser().getId());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: I think this line is too long. Did you apply the formatting rules?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 8]
```diff
@@ -46,16 +50,36 @@
...
+                key = ConfigCore.getParameter("language.force-default", user.getLanguage());
+            } catch (DAOException e) {
+                logger.error("Error in retrieving user : " + e.getMessage());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: In other places users get informed that there is an error. Why not here?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 9]
```diff
@@ -135,6 +161,15 @@ public void switchLanguage(String langCodeCombined) throws IOException {
         } else {
             locale = new Locale(languageCode[0]);
         }
+        try {
+            User user = serviceManager.getUserService().getById(serviceManager.getUserService().getAuthenticatedUser().getId());
+            user.setLanguage(locale.toString());
+            serviceManager.getUserService().save(user);
+        } catch (DataException e) {
+            logger.error("Error in saving user : " + e.getMessage());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: In other places users get informed that there is an error. Why not here?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 10]
```diff
@@ -135,6 +161,15 @@ public void switchLanguage(String langCodeCombined) throws IOException {
...
+        } catch (DAOException e) {
+            logger.error("Error in retrieving user" + e.getMessage());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: In other places users get informed that there is an error. Why not here?

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 11]
```diff
@@ -46,16 +50,36 @@
...
+
+    /**
+     * Set session locale field id.
+     *
+     *
+     */
+    public void setSessionLocaleFieldId() {
+        String key = "";
+        if (Objects.isNull(serviceManager.getUserService().getAuthenticatedUser())) {
+            key = ConfigCore.getParameter("language.force-default", "de");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Please use or define a new key like in PR #1623 and do same for default value.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java [hunk 12]
```diff
@@ -46,16 +50,36 @@
...
+        } else {
+            try {
+                User user = serviceManager.getUserService().getById(serviceManager.getUserService().getAuthenticatedUser().getId());
+                key = ConfigCore.getParameter("language.force-default", user.getLanguage());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Please use or define a new key like in PR #1623.

**Comment by henning-gerhardt** [comment]
There is a merge conflict in

-  Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java

Please solve this conflict while you do other needed changes. Thank you.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java (hunk 6 above)
Comment (IkramMaalej) [reply Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Yes, I did.
It 's not too long according to the CheckStyle.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java (hunk 10 above)
Comment (IkramMaalej) [reply Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Yes, I did.
It 's not too long according to the CheckStyle.

File: Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java (hunk 6 above)
Comment (henning-gerhardt) [reply Kitodo/src/main/java/de/sub/goobi/forms/SpracheForm.java]: Checkstyle allow lines with 140 characters and this line is below. So it should be ok for now.

**Comment by henning-gerhardt** [comment]
Selenium tests on Travis CI are failing on two selenium tests:

- AddingST.addDocketTest
- AddingST.addProjectTest

File: Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java [hunk 13]
```diff
@@ -43,14 +43,6 @@
     @FindBy(className = "ui-selectonemenu-trigger")
     private WebElement selectTrigger;
 
-    @SuppressWarnings("unused")
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java]: Why did you remove this and other lines in this file? This does not look right for me as you did not change anything related to project editing.

Comment (IkramMaalej) [reply Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java]: I removed this lines because there is no need to add explicitly the startDateInput and the endDateInput for tests and instead of that, I used implicitly the default input values of this dates (today as default value) which are already entered automatically in the form  and this is why I didn't change any other thing.
Deleting this lines and using the default values of dates eliminate the problem of  date format when the UI is in English(08/05/18) or in German("05.08.19").

Comment (henning-gerhardt) [reply Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java]: I'm not sure why this was added by @Beacze. I think there is a reason for it. Maybe @Beacze can answer if this could really be removed or solved in an other way?

Comment (Beacze) [reply Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java]: I have discussed this with @IkramMaalej. I need to check what exactly causes test fails after she adds possiblity for user to manually change the language. Until then this part of code can be commented out / removed.

Comment (henning-gerhardt) [reply Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectEditPage.java]: Thank you @Beacze.
